### PR TITLE
Date selection

### PR DESCRIPTION
**Pull request by ScienceMaster7**
What I did : 

- I made the cards a component. 
- I made the content on the cards a component.
- I created a component for the selection of a month.
- I created a component for the selection of a year.
- I implemented the logic for selecting a month and display the corresponding streak item.
-  I implemented the logic for selecting a year and display the corresponding streak item.

The dummy data is there to demonstrate how it works.

I know that the inactive days after the last active day are not displayed, but I will work on that.

Constructive feedback is appreciated !

### CODE REVIEW

File: src/components/CardContent.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  const currentMonthDays = datesCurrentMonth.map((date) => {
+    return date[0];
+  });
+
+  const [days, setDays] = useState(currentMonthDays);
+  const [availableMonths, setAvailableMonths] = useState();
+  const [monthSelection, setMonthSelection] = useState(false);
+  const [availableYears, setAvailableYears] = useState();
+  const [yearSelection, setYearSelection] = useState(false);
```
Comment (patrikglanemann) [inline src/components/CardContent.js:43]: For readability you could put all states and variables to the top, just an idea:)

File: src/components/CardContent.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  function onClickHandleMonth() {
+    if (yearSelection === true) {
+      toast("You can't select a month while selecting a Year");
```
Comment (patrikglanemann) [inline src/components/CardContent.js:47]: This pop-up message (and the others) is super useful for me as a user!

**Review by patrikglanemann — COMMENTED** [review]
Awesome work so far! It becomes a very complex app with more and more features👍

File: src/components/CardContent.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        if (!months.includes(date[1])) {
+          months.push(date[1]);
+        }
+      }
+      setAvailableMonths(months);
+      setMonthSelection(true);
+    }
+  }
+
+  function monthCallback(newMonth) {
```
Comment (mpagels) [inline src/components/CardContent.js:65]: What is `monthCallback` doing? Maybe find a better descriptive name for the function.

File: src/components/SelectMonths.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export default function SelectMonths({ monthNames, months, callBack }) {
+  function handleOnClickNewMonth(event) {
+    const month = event.target.value;
+    callBack(month);
```
Comment (mpagels) [inline src/components/SelectMonths.js:4]: What is `callBack` doing with `month`. Please find a better name. You will thank yourself when you come back to the project month later.

File: src/components/SelectMonths.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  let renderedButtons = [];
+  for (let i = 0; i < months.length; i++) {
+    const monthButton = (
+      <button onClick={handleOnClickNewMonth} key={i} value={months[i]}>
+        {monthNames[months[i]]}
+      </button>
+    );
+    renderedButtons.push(monthButton);
+  }
```
Comment (mpagels) [inline src/components/SelectMonths.js:15]: Try to use modern JavaScript to write a loop. It is not wrong though, but a map is way more readable. And you're saving a lot of lines. ;-)
```suggestion
  let renderedButtons = months.map((month, index) => (
      <button onClick={handleOnClickNewMonth} key={index} value={month}>
        {monthNames[month]}
      </button>))
```

File: src/components/SelectYears.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+  let renderedButtons = [];
+  for (let i = 0; i < years.length; i++) {
+    const monthButton = (
+      <button onClick={handleOnClickNewYear} key={i} value={years[i]}>
+        {years[i]}
+      </button>
+    );
+    renderedButtons.push(monthButton);
+  }
```
Comment (mpagels) [inline src/components/SelectYears.js:14]: Same as in `SelectedMonth` component. Try to use modern ECMAScript.

**Pull request merged**